extensions/wikia/Captcha/Module/BaseCaptcha.class.php

```diff
@@ -0,0 +1,235 @@
+
+	public function getCaptcha() {
+		$a = mt_rand( 0, 100 );
+		$b = mt_rand( 0, 10 );
+
+		/* Minus sign is used in the question. UTF-8,
+		   since the api uses text/plain, not text/html */
+		$op = mt_rand( 0, 1 ) ? '+' : '−';
+
+		// No space before and after $op, to ensure correct
+		// directionality.
+		$test = "$a$op$b";
+		$answer = ( $op == '+' ) ? ( $a + $b ) : ( $a - $b );
+		return [
+			'question' => $test,
+			'answer' => $answer
+		];
+	}
+
+	public function addCaptchaAPI( &$resultArr ) {
+		$captcha = $this->getCaptcha();
+		$index = $this->storeCaptcha( $captcha );
+		$resultArr['captcha']['type'] = 'simple';
+		$resultArr['captcha']['mime'] = 'text/plain';
+		$resultArr['captcha']['id'] = $index;
+		$resultArr['captcha']['question'] = $captcha['question'];
+	}
+
+	/**
+	 * Insert a captcha prompt into the edit form.
+	 * This sample implementation generates a simple arithmetic operation;
+	 * it would be easy to defeat by machine.
+	 *
+	 * Override this!
+	 *
+	 * @return string HTML
+	 */
+	public function getForm() {
+		$captcha = $this->getCaptcha();
+		$index = $this->storeCaptcha( $captcha );
+
+		return "<p><label for=\"wpCaptchaWord\">{$captcha['question']}</label> = " .
+		\Xml::element( 'input', [
+			'name' => 'wpCaptchaWord',
+			'id'   => 'wpCaptchaWord',
+			'tabindex' => 1,
+		] ) . // tab in before the edit textarea
+		"</p>\n" .
+		\Xml::element( 'input', [
+			'type'  => 'hidden',
+			'name'  => 'wpCaptchaId',
+			'id'    => 'wpCaptchaId',
+			'value' => $index,
+		] );
+	}
+
+	/**
+	 * Show a message asking the user to enter a captcha on edit
+	 * The result will be treated as wiki text
+	 *
+	 * @param string $action Action being performed
+	 *
+	 * @return string
+	 */
+	public function getMessage( $action ) {
+		$name = 'captcha-' . $action;
+		$text = wfMsg( $name );
+		# Obtain a more tailored message, if possible, otherwise, fall back to
+		# the default for edits
+		return wfEmptyMsg( $name, $text ) ? wfMsg( 'captcha-edit' ) : $text;
+	}
+
+	/**
+	 * Check if the submitted form matches the captcha session data provided
+	 * by the plugin when the form was generated.
+	 *
+	 * Override this!
+	 *
+	 * @param string $answer
+	 * @param array $info
+	 *
+	 * @return bool
+	 */
+	public function keyMatch( $answer, $info ) {
+		return $answer == $info['answer'];
+	}
+
+	/**
+	 * @param \ApiBase $module
+	 * @param array $params
+	 *
+	 * @return bool
+	 */
+	public function APIGetAllowedParams( &$module, &$params ) {
+		if ( !$module instanceof \ApiEditPage ) {
+			return true;
+		}
+		$params['captchaword'] = null;
+		$params['captchaid'] = null;
+
+		return true;
+	}
+
+	/**
+	 * @param \ApiBase $module
+	 * @param array $desc
+	 * @return bool
+	 */
+	public function APIGetParamDescription( &$module, &$desc ) {
+		if ( !$module instanceof \ApiEditPage ) {
+			return true;
+		}
+		$desc['captchaid'] = 'CAPTCHA ID from previous request';
+		$desc['captchaword'] = 'Answer to the CAPTCHA';
+
+		return true;
```

Comment:
Same simplifying logic could be used here. It's personal preference, but I find it nice to consolidate return statements if possible.